pkg/machines/components/vmSnapshotsCreateModal.jsx

```diff
@@ -0,0 +1,165 @@
+import {
+    Button,
+    TextArea,
+    TextInput
+} from "@patternfly/react-core";
+
+import { ModalError } from "cockpit-components-inline-notification.jsx";
+import { createSnapshot } from "../libvirt-dbus.js";
+import { getVmSnapshots } from '../actions/provider-actions.js';
+
+const _ = cockpit.gettext;
+
+const NameRow = ({ onValueChanged, dialogValues, onValidate }) => {
+    return (
+        <>
+            <label className="control-label" htmlFor="name">
+                {_("Name")}
+            </label>
+            <FormGroup validationState={dialogValues.validationError.name ? "error" : undefined} controlId="name">
+                <TextInput value={dialogValues.name}
+                    id="name"
+                    type="text"
+                    onChange={(value) => onValueChanged("name", value)}
+                    onBlur={() => onValidate("name")}
+                    aria-label={_("Name input text")}
+                />
+                { dialogValues.validationError.name &&
+                <HelpBlock>
+                    <p className="text-danger">{dialogValues.validationError.name}</p>
+                </HelpBlock> }
+            </FormGroup>
+
+        </>
+    );
+};
+
+const DescriptionRow = ({ onValueChanged, dialogValues }) => {
+    return (
+        <>
+            <label className="control-label" htmlFor="description">
+                {_("Description")}
+            </label>
+            <TextArea value={dialogValues.description}
+                id="description"
+                placeholder={_("Optional arbitrary description of a snapshot")}
+                onChange={(value) => onValueChanged("description", value)}
+                resizeOrientation="vertical"
+                aria-label={_("Description input text")}
+            />
+        </>
+    );
+};
+
+export class CreateSnapshotModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: props.vm.name + '_' + moment().format("YYYY-MM-DD_HH:mm"),
+            description: "",
+            validationError: {},
+        };
+
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+        this.onValidate = this.onValidate.bind(this);
+        this.onCreate = this.onCreate.bind(this);
+    }
+
+    onValueChanged(key, value) {
+        this.setState({ [key]: value });
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    onValidate(subject) {
+        const { name, validationError } = this.state;
+        const { vm } = this.props;
+
+        if (subject === "name") {
+            const newValidationError = { ...validationError };
+            if (vm.snapshots.findIndex(snap => snap.name === name) > -1)
+                newValidationError.name = "Snapshot with this name already exists";
```

Comment:
I would prefer the shorter _name already exists_.
Certain languages, such as German, has a tendency to get very long in translations.
It is also clear from the context that this is about snapshots.